### PR TITLE
Upgrade to v4.6.0 of the AWS Terraform Provider

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #1821
Fixes: #1828
Fixes: #1842
Fixes: #1847
Fixes: #1857
Fixes: #1865
Fixes: #1876﻿

### CODE REVIEW

File: provider/resources.go [hunk 1]
```diff
@@ -1326,22 +1335,24 @@ func Provider() tfbridge.ProviderInfo {
...
-			"aws_vpn_gateway_route_propagation":          {Tok: awsResource(ec2Mod, "VpnGatewayRoutePropagation")},
-			"aws_ec2_traffic_mirror_filter":              {Tok: awsResource(ec2Mod, "TrafficMirrorFilter")},
-			"aws_ec2_traffic_mirror_filter_rule":         {Tok: awsResource(ec2Mod, "TrafficMirrorFilterRule")},
-			"aws_ec2_traffic_mirror_session":             {Tok: awsResource(ec2Mod, "TrafficMirrorSession")},
-			"aws_ec2_traffic_mirror_target":              {Tok: awsResource(ec2Mod, "TrafficMirrorTarget")},
-			"aws_ec2_availability_zone_group":            {Tok: awsResource(ec2Mod, "AvailabilityZoneGroup")},
+			"aws_vpc_endpoint_connection_notification":    {Tok: awsResource(ec2Mod, "VpcEndpointConnectionNotification")},
+			"aws_vpc_endpoint_route_table_association":    {Tok: awsResource(ec2Mod, "VpcEndpointRouteTableAssociation")},
+			"aws_vpc_endpoint_service":                    {Tok: awsResource(ec2Mod, "VpcEndpointService")},
+			"aws_vpc_endpoint_service_allowed_principal":  {Tok: awsResource(ec2Mod, "VpcEndpointServiceAllowedPrinciple")},
```
Comment (guineveresaenger) [inline provider/resources.go:1341]: It's been this way for a bit apparently but this should be `VpcEndpointServiceAllowedPrincipal` to map correctly, no? Not sure how hard it is to fix after the fact

File: provider/resources.go [hunk 2]
```diff
@@ -2072,23 +2100,34 @@ func Provider() tfbridge.ProviderInfo {
...
+			// Network manager
+			"aws_networkmanager_connection":                               {Tok: awsResource(networkManagerMod, "Connection")},
+			"aws_networkmanager_customer_gateway_association":             {Tok: awsResource(networkManagerMod, "CustomerGatewayAssociation")},
+			"aws_networkmanager_device":                                   {Tok: awsResource(networkManagerMod, "Device")},
+			"aws_networkmanager_global_network":                           {Tok: awsResource(networkManagerMod, "GlobalNetwork")},
+			"aws_networkmanager_link":                                     {Tok: awsResource(networkManagerMod, "Link")},
+			"aws_networkmanager_link_association":                         {Tok: awsResource(networkManagerMod, "LinkAssociation")},
+			"aws_networkmanager_site":                                     {Tok: awsResource(networkManagerMod, "Site")},
+			"aws_networkmanager_transit_gateway_connect_peer_association": {Tok: awsResource(networkManagerMod, "TransitGatewayConnectPeerAssociation")},
+			"aws_networkmanager_transit_gateway_registration":             {Tok: awsResource(networkManagerMod, "TransitGatewayRegistry")},
```
Comment (guineveresaenger) [inline provider/resources.go:2112]:
```suggestion
			"aws_networkmanager_transit_gateway_registration":             {Tok: awsResource(networkManagerMod, "TransitGatewayRegistration")},
```

File: provider/resources.go [hunk 3]
```diff
@@ -4072,6 +4116,8 @@ func Provider() tfbridge.ProviderInfo {
 			"aws_batch_compute_environment": {Tok: awsDataSource(batchMod, "getComputeEnvironment")},
 			"aws_batch_job_queue":           {Tok: awsDataSource(batchMod, "getJobQueue")},
 			"aws_batch_scheduling_policy":   {Tok: awsDataSource(batchMod, "getSchedulingPolicy")},
+			"aws_backup_framework":          {Tok: awsDataSource(batchMod, "getFramework")},
```
Comment (guineveresaenger) [inline provider/resources.go:4119]:
```suggestion
			"aws_backup_framework":          {Tok: awsDataSource(backupMod, "getFramework")},
```

File: provider/resources.go [hunk 4]
```diff
@@ -4072,6 +4116,8 @@ func Provider() tfbridge.ProviderInfo {
...
+			"aws_backup_report_plan":        {Tok: awsDataSource(batchMod, "getReportPlan")},
```
Comment (guineveresaenger) [inline provider/resources.go:4120]:
```suggestion
			"aws_backup_report_plan":        {Tok: awsDataSource(backupMod, "getReportPlan")},
```

**Review by guineveresaenger — CHANGES_REQUESTED** [review]
Found a couple discrepancies in the mappings, but TYSM for this beast!

File: provider/resources.go [hunk 5]
```diff
@@ -696,6 +699,8 @@ func Provider() tfbridge.ProviderInfo {
 			"aws_backup_vault_policy":             {Tok: awsResource(backupMod, "VaultPolicy")},
 			"aws_backup_vault_lock_configuration": {Tok: awsResource(backupMod, "VaultLockConfiguration")},
 			"aws_backup_region_settings":          {Tok: awsResource(backupMod, "RegionSettings")},
+			"aws_backup_framework":                {Tok: awsResource(backupMod, "Framework")},
+			"aws_backup_report_plan":              {Tok: awsResource(backupMod, "Report")},
```
Comment (jkodroff) [inline provider/resources.go:703]: Should either be `ReportPlan` or have a comment indicating why it's `Report`.

**Review by jkodroff — CHANGES_REQUESTED** [review]

File: provider/resources.go (hunk 1 above)
Comment (stack72) [reply provider/resources.go:1341]: I am going to say we don't need to fix this in this release

**Review by jkodroff — APPROVED** [review]

**Pull request merged**